pandas/core/dtypes/common.py

```diff
@@ -304,34 +304,6 @@ def is_offsetlike(arr_or_obj):
     return False
 
 
-def is_period(arr):
-    """
-    Check whether an array-like is a periodical index.
-
-    Parameters
-    ----------
-    arr : array-like
-        The array-like to check.
```

Comment:
unfortunately I think we need to deprecate this (and remove in next version)

Reply:
isn't this change orthogonal to this PR?

Reply:
> isn't this change orthogonal to this PR?

Sure, can be separated out.  This started out as a grab-bag PR.

> unfortunately I think we need to deprecate this (and remove in next version)

Darn, OK.